react/features/app/actions.js

```diff
@@ -147,15 +147,15 @@ function _navigate(state) {
  */
 function _setRoomAndNavigate(newRoom) {
     return (dispatch, getState) => {
-        const oldRoom = getState()['features/base/conference'].room;
-
         dispatch(setRoom(newRoom));
 
         const state = getState();
-        const room = state['features/base/conference'].room;
+        const { app } = state['features/app'];
+        const newRoute = _getRouteToRender(state);
 
-        if (room !== oldRoom) {
-            _navigate(state);
+        if (!areRoutesEqual(newRoute, currentRoute)) {
```

Comment:
`RouteRegistry` clones routes before it exposes them from its internal state so `areRoutesEqual` would probably better fit as a (static?) method of `RouteRegistry`.